presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -2280,6 +2286,11 @@ public static DataSize getFilterAndProjectMinOutputPageSize(Session session)
         return session.getSystemProperty(FILTER_AND_PROJECT_MIN_OUTPUT_PAGE_SIZE, DataSize.class);
     }
 
+    public static boolean isMaterializeAllCtes(Session session)
```

Comment:
Instead of having a boolean session property, perhaps we can have a enum to offer more flexibility, like rename the property to CTE_MATERIALIZE_STRATEGY and offer ALL, COST_BASED, NONE etc options. So that in the end, we do not need to introduce another session property when introducing cost based materialization strategy.

Reply:
Makes sense

Reply:
Better to return the strategy, and have check in optimizer, rather than do the boolean check here. You can refer to other similar session properties as a reference.